server/routerlicious/packages/services-core/src/lambdas.ts

```diff
@@ -165,6 +165,16 @@ export interface IPartitionLambdaConfig {
 	documentId: string;
 }
 
+/**
+ * Whether the boxcar message includes the optional Routing Key fields.
+ * @internal
+ */
+export function isCompleteBoxcarMessage(
+	boxcar: IBoxcarMessage,
+): boxcar is Required<IBoxcarMessage> {
+	return boxcar.documentId !== undefined && boxcar.tenantId !== undefined;
```

Comment:
also curious what if docId or tenantId is "", would ("" !== undefined) be true?

Reply:
`("" !== undefined) === true`, but the point here is that there are some cases where documentId and tenantId are explicitly set as `undefined` when there is an error or missing information in a message. This type guard protects against that scenario.

In this case, `documentId: ""` is considered valid, which is intentional, because the scope of this PR is adding `null` and `undefined` protections. I do not know what the implications would be of blocking `""` here.

Reply:
>  the scope of this PR is adding null and undefined protections
I see, make sense :)